hc/front/tests/test_add_shopify.py

```diff
@@ -0,0 +1,112 @@
+from hc.test import BaseTestCase
+from shopify_trois import Credentials, Collection
+from shopify_trois.models import Shop
+from shopify_trois.engines.http import Json as Shopify
+from django.core.urlresolvers import reverse
+import shopify
+import requests
+from hc.api.models import Check
+
+class AddShopifyAlertTestCase(BaseTestCase):
+    """This class contains tests to handle adding checks"""
+
+
+    def test_it_redirects_add_shopify(self):
+        """test that when the correct authentication keys are given that it returns valid response """
+
+        self.client.login(username="alice@example.org", password="password")
+        response = self.client.get( "/integrations/add_shopify/")
+
+        assert response.status_code == 200
+
+    def test_it_accepts_connection_to_shopify(self):
+        """test that when the correct authentication keys are given that it returns valid response """
+        API_KEY = "2ea13bd64c9c06f5ff5501dd6872ecda"
+
+        PASSWORD = "d602f072d114aceca3c21a2234582ce3"
+
+        EVENT = "order/create"
+
+        NAME = "Create Order"
+
+        SHOP_NAME = "Duuka1"
+
+        form = {"api_key": API_KEY,
+                "password": PASSWORD, "event": EVENT, "name": NAME, "shop_name": SHOP_NAME}
+
+        url = reverse("hc-create-shopify-alerts")
+
+        self.client.login(username="alice@example.org", password="password")
+        response = self.client.post(url, form)
+        
+        self.assertRedirects(response, "/checks/")
+
+        assert response.status_code == 302
+
+    def test_it_doesnot_accept_wrong_details(self):
+        API_KEY = "2ea13bd64c9c06f5ff5501dd6872ecda"
```

Comment:
Inclusion of the API key and password information in this tests poses a security risk to the application . I suggest you store them in the system environment instead.